### DISCUSSION TITLE
Data fetching strategy getStaticProps

### DISCUSSION THREAD

**Discussion by pseudoeazy** [vercel/community#503] [Help]
Nextjs recommends getStaticPaths and getStaticProps for dynamic routing for blog pages.

getStaticPaths needs to return an ID for getStaticProps to fetch the required data.

That seems like double query if I get my data from a different server.

I don't want to include query parameters to my url. 

How to I implement data fetching for  dynamic routes using with getStaticProps by blogData ID?

I don't want to use getServerProps too, since it is a blog page and getServerProps regenerates its HTML on each request.

**Comment by pseudoeazy** [comment]
I stumbled across something that solves my problem.

Nextjs Rewrite rules in next.config.js

**Reply by ismaelrumzan** [reply]
Hi there!
I'm glad you found an answer. I'm just curious related to your concern about `double query` when using a dynamic path. Are you concerned that for each post page, there will be a request for the page and a request to the database to generate the page?
All the paths i.e. post ids you specific in `GetStaticPaths` will be used to generate a separate static page at build time. Therefore, there will be only one request for each of those pages at `runtime`.
This is an [example](https://github.com/vercel/next.js/blob/canary/examples/cms-wordpress/pages/posts/%5Bslug%5D.js)

**Reply by pseudoeazy** [reply]
Thank you for your help.

This was my concern

1. A url(example.com/my-post-title/?id=1) GET request is made to the page.

2. GET request to server to fetch data with ID is made and data is retrieved in getStaticProps by ID.

3. I don't want the URL to display (example.com/my-post-title/?id=1) rather it should display (example.com/my-post-title) ID should be hidden on the page no matter the requested  ID query parameter passed along with the URL.

It is basically a pretty URL issue.
